### DISCUSSION TITLE
Shelly 2.5 temperature impacted by wifi access point 1 vs 2?

### DISCUSSION THREAD

**Discussion by rsteininger** [arendst/Tasmota#11831] [Q&A] [answered]
hi there, 
I have a rather interesting phenomenon. I am using a (spare) shelly 2.5 to power monitor our boiler on one channel and the house water pump on the other channel. The day before yesterday for the first time, both cut out and when I looked into it, it was caused by the shelly 2.5 which switched off due to over temperature. While the installation location is anything but good (behind a switch in the wall covered by an enclosure where the boiler is installed. And thus especially during the winter time, very warm environment. I would guess about 35 DegC or higher ambient temperature). Now, this has been running for more then a year without a hitch. However, when I now look back over temperature curve over the last 6 months, I see a "strange" jump up and down of the average module temperature. While the active running of the water pump especially and the boiler to some extent have an impact on the module temperature, there seems to be another "outside" driving force influencing the "base line" temperature. Here a screenshot of the trend curve:
![image](https://user-images.githubusercontent.com/5363948/115611623-3591fa80-a2e2-11eb-9859-8283b67a6510.png)
You can see that there are periods of "low temperature" (~68 DegC?) and high temperature (~76 DegC?). Nothing has changed during these time with regards to use of the heating, the water pump or any programming associated with the shelly! 
However, I can loosely align the changes between these phases to issues with the home network, e.g. the internet router went down which provides one of the two configured access points for the shelly 2.5! 
As this kind of dawned on me over the last two days, I manually triggered a change of the access point today (ap 0). The duration is not yet long enough but it seems to have the impact I was suspecting! 
However - and here comes the "funny" and strange part - the configured access point 1, which is closer and provides a stronger signal seem to cause the higher temperature in the shelly (-> ~76 DegC), while when I switch to the other wifi access point, which is better equipment but further away resulting in a signal drop from about 65% (ap1) down to 44% (ap2). 
If anything, I would have expected the opposite that a better signal would cause less power requirement in the shelly and thus less internal heating! 
Does anyone have an explanation for what I seem to have experienced? Is this possibly by design or could there be something else at force? I have not yet tried replacing AP1 with other hardware to see if I would get similar results as with AP2. 
Any thoughts anyone?
THANKS
Robert

**Comment by sfromis** [comment, marked as answer]
According to Alterco, the ambient temperature range is between -40°C up to 40°C, sounds like you may be outside of that. Tasmota over-temperature protection is controlled by [SetOption42](https://tasmota.github.io/docs/Commands/#setoption42), default 90. And the device is known to run pretty hot, depending on load up to 90°C (according again to Alterco), which I assume is within max ambient temperature range. I've not followed temperatures much, but a quick count indicate that each of my Shelly 2.5 device tend to vary within a 10°C range, and a few times a bit above that.

**Reply by rsteininger** [reply]
Thanks! I know I am pushing the limit of the device but that was not the issue I was reporting! 
My device temperature changes depending on whether I am connected to the configured wifi Access Point 1 (stronger wifi, but module average temperature rises to ~76 DegC in idle) and Access Point 2 (weaker wifi, but better device hardware, module average temperature drops to ~68 DegC in idle). 
I have no rational explanation why the selection of the wifi access point influences the internal Shelly module temperature. 
(is it a feature? is it a bug? is it due to issues with the wifi signal from AP1 [it is ISP provided hardware but fairly recent build]).
